rigetti/qcs-sdk-rust
# feat(python): add type hints to qcs-sdk-python

Resolves #156 

## Example

![image](https://user-images.githubusercontent.com/4324359/191145521-987a99b9-7d1b-4518-9653-cddde07a7d68.png)

## Implementation

Per the [pyo3 docs](https://pyo3.rs/v0.16.4/python_typing_hints.html#if-you-do-not-have-other-python-files): All that we need for our use case is to define a single `pyi` file for `qcs_sdk`. `maturin` detects the file and does the rest of the work for us. 

## How to test
1. `cd crates/python && poetry install`
2. In a virtualenv run `maturin develop` to build and install the bindings. `maturin` should output `Found type stub file at qcs_sdk.pyi`.
3. In your LSP enabled editor of choice, `import qcs_sdk` and start exploring the type hints.

To make sure the hints would be distributed through CI, I also built and installed the wheel for my platform just as the CI workflow would, then verified that the type hints were available. To try that yourself:

1. Activate some virtualenv then from the root of the project:
2. `maturin build -i python --release --manifest-path crates/python/Cargo.toml --out dist --sdist`
3. `pip install qcs-sdk-python --no-index --find-links dist --force-reinstall`
4. In your LSP enabled editor of choice, `import qcs_sdk` and start exploring the type hints.
